promql/parser/ast.go

```diff
@@ -119,12 +119,24 @@ type Call struct {
 	PosRange posrange.PositionRange
 }
 
+type MatrixSelectorBinOp struct {
+	Op  ItemType
+	Num *NumberLiteral
+}
+
+type MatrixSelectorBinOps struct {
+	Ops []*MatrixSelectorBinOp
+
+	PosRange posrange.PositionRange
+}
+
 // MatrixSelector represents a Matrix selection.
 type MatrixSelector struct {
 	// It is safe to assume that this is an VectorSelector
 	// if the parser hasn't returned an error.
 	VectorSelector Expr
```

Comment:
Since this is an Expr already, we could stuff a BinOp here too I think, but that would be a breaking change.

Reply:
I would like BinOp here most I think - It makes this nicely forwards compatible to generalizing this to arbitrary Exprs on RHS too :thinking: